tests/test_vm.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from dotenv import load_dotenv
+import os
+from src.ssh_client import SSHClient
+from src.winrm_client import WinRMClient
+from src.utils import calculate_checksum
+
+load_dotenv()
+
+@pytest.fixture
+def shell_client(request):
+    os_option = request.param
+    if os_option == 'linux':
+        host = os.getenv('LINUX_HOST')
+        user = os.getenv('LINUX_USER')
+        password = os.getenv('LINUX_PASSWORD')
+        return SSHClient(host, user, password)
+    elif os_option == 'windows':
+        host = os.getenv('WINDOWS_HOST')
+        user = os.getenv('WINDOWS_USER')
+        password = os.getenv('WINDOWS_PASSWORD')
+        return WinRMClient(host, user, password)
+    else:
+        pytest.fail("Please provide a valid OS option: --os=linux or --os=windows")
+
+@pytest.fixture
+def directory(request):
+    dirpath = request.config.getoption("--dirpath")
+    if not dirpath:
+        pytest.fail("Please provide a directory path using --dirpath option.")
+    return dirpath
+
+@pytest.mark.parametrize("shell_client", ["linux", "windows"], indirect=True)
+def test_create_directory(shell_client, directory, request):
+    os_option = 'linux' if isinstance(shell_client, SSHClient) else 'windows'
+    os_types = request.config.getoption("--os").split(",")
+
+    if os_option not in os_types:
+        pytest.skip(f"Skipping test for OS type {os_option}")
+
+    # Determine the absolute path of the scripts
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+    if isinstance(shell_client, SSHClient):
+        local_script = os.path.join(base_dir, '../scripts/create_dir.sh')
```

Comment:
instead of using prepared scripts please implement function to create file. In order to do not hardcode directory path and make script os independent, you can use tempfile.TemporaryDirectory() context manager